### ISSUE TITLE
[DocDB][xCluster-Atomicity] Atomicity Inconsistency and XCluster safe time not yet initialized for namespace ERROR at Target side.

### ISSUE THREAD

**Issue by Arjun-yb** [yugabyte/yugabyte-db#15269]
Jira Link: [DB-4432](https://yugabyte.atlassian.net/browse/DB-4432)
### Description

Steps:
1. Create 2 universes(Source and Target) with below GFLAGS in both universes
```
Master:
{
"enable_replicate_transaction_status_table": true,
"tablet_split_high_phase_shard_count_per_node": 10000, 
"tablet_split_high_phase_size_threshold_bytes": 10485760, 
"tablet_split_low_phase_size_threshold_bytes": 1048576, 
"tablet_split_low_phase_shard_count_per_node": 48,
"tablet_split_limit_per_table": 512
}

Tserver:
{
"ysql_num_shards_per_tserver":1
}
```
2. Create database and tables at both sides
```
create table table1(id int primary key, description text);
create table table2(id int primary key, description text);
```
4. Setup replication from Source to Target
5. Change the xCluster role at Target
`yb-admin -master_addresses <master-address> change_xcluster_role STANDBY`
6. Load data with transactional workload(200 writes/transaction - >1000 transactions) 
7. Observe the tablet splitting, Once splitting happens then validate data
8. Observed `XCluster safe time not yet initialized for namespace`

Then

9. Create one more DB at both sides and 2 more tables
```
create table table1(id int primary key, description text);
create table table2(id int primary key, description text);
```
10. Setup replication and run select query at target and observe same error(`XCluster safe time not yet initialized for namespace`)
11. Wait for some time and again run select query at target - now it works
12. Then Load data with transactional workload(200 writes/transaction - >1000 transactions) at source and validate data at both sides

**Observed**
1. Replication didn’t happen immediately from source to target(after Step#12) .
2. Replication was very slow and observed inconsistent atomicity.
3. After some time(when splitting started for the tables(after Step#12) again observed `XCluster safe time not yet initialized for namespace`

**Comment by Arjun-yb** [comment]
get_xcluster_safe_time result
```
[
    {
        "namespace_id": "0000400b000030008000000000000000",
        "safe_time": "2112-09-17 23:53:47.370495",
        "safe_time_epoch": "4503599627370495"
    },
    {
        "namespace_id": "00004000000030008000000000000000",
        "safe_time": "2112-09-17 23:53:47.370495",
        "safe_time_epoch": "4503599627370495"
    },
    {
        "namespace_id": "00000000000000000000000000000001",
        "safe_time": "2022-12-08 13:44:42.691452",
        "safe_time_epoch": "1670507082691452"
    }
]
```

**Comment by lingamsandeep** [comment]
Fixed by 64a7b3aa96894604689ed7f7a1f74e52078b1c57.
